src/controllers/products.js

```diff
@@ -22,6 +22,22 @@ exports.createProduct = async (req, res, next) => {
   return res.status(201).json({ success: true, data: product });
 };
 
+// check if requestedBuyers doesn't have buyerId already. product.requestedBuyers.find(i => i.toHexString() === buyerId.toHexString())
+exports.requestProduct = async (req, res, next) => {
+  const productId = req.params.id;
+  const buyerId = req.user._id;
+  const product = await Product.findOne({ _id: productId });
```

Comment:
Perhaps you want to add another check, if the product doesn't belong to the user!
Imagine if the user posted a product, then tried to request it, we shouldn't allow that.